tensornetwork/network_operations.py

```diff
@@ -817,3 +833,86 @@ def get_neighbors(node: AbstractNode) -> List[AbstractNode]:
         neighbors.append(edge.node1)
         neighbors_set.add(edge.node1)
   return neighbors
+
+
+def nodes_to_json(nodes: List[AbstractNode]) -> str:
+  """
+  Create a JSON string representing the Tensor Network made up of the given 
+  nodes. Nodes and their attributes, edges and their attributes and tensor
+  values are included.
+  
+  Tensors are serialized according the the format used by each tensors backend.
+  
+  For edges spanning included nodes and excluded nodes the edge attributes are 
+  preserved in the serialization but the connection to the excluded node is 
+  dropped. The original edge is not modified.
```

Comment:
I also realized I will want to mark certain edges as special int the network. (For example, labeling the input vs output legs of an operator network.) How should we go about doing that?

Reply:
I think depends on how this information is used. My first inclination would be to have this as an attribute of the edge that is saved in the `to_serial_dict` method, and can be specified as a keyword argument in an Edge constructor.

Reply:
I think either way just save it as an attribute in `to_serial_dict` and if there is some semantic where it can't be given to the Edge constructor then we can add a `from_serial_dict` method to the Edge class